packages/embark/src/lib/modules/blockchain_process/gethClient.js

```diff
@@ -73,6 +73,10 @@ class GethClient {
       cmd.push("--syncmode=" + config.syncMode);
     }
 
+    if(this.isTestNetwork(config)) {
+      cmd.push("--gcmode=archive");
```

Comment:
Should this be made an option in `config/blockchain.js`?

Reply:
👍

Reply:
I don't think this should be an option, as the setting is completely detached from functionality, ie. it's not obvious that archival mode will lead to debugging older transactions.

I'd say we should probably tighten the network IDs instead to just do this on local testnet.

Reply:
Fair enough. It just might be that some time in the future a user of Embark wishes to run on testnet without this option and then they can't because we've burned it into Embark's DNA.

But maybe chances are very low this will happen. So we can go with this for now and do as you suggested.

Reply:
I agree with all of you guys.

The solution IMO, is to advertise it as it is. So we add `gcmode` to the config. By default, we don't put it there (just add it to the docs). I just looked, and in Parity, this is named `pruning`.  So we might need to find a name that works for both. `pruning` might be better, because it is referenced a bit in Geth too.

So, if the config option is set, we use that, otherwise, we check if it's a test network.

This way, we don't allude that this is a way to debug, just a way to enable archive mode.